### ISSUE TITLE
Issue with downloading image file

### ISSUE THREAD

**Issue by nhatmvo** [fsouza/fake-gcs-server#429]
After I uploaded a file into server successfully, when downloaded that file using curl, I cannot open that file (the file is damaged or corrupted). I even used `docker cp` to move that file from `storage/<my-bucket>` within fake-gcs-server container into my host machine, but I still cannot open it neither.
Is this intended or my request is missing something?

here is my request:

```
curl --location --request POST 'https://0.0.0.0:4443/upload/storage/v1/b/sample-bucket/o?uploadType=media&name=tux.png' \
--header 'Content-Type: image/png' \
--data-binary '@/Users/nhatvm/Downloads/Tux.png'
```
and download it using:

```
curl -X GET 
  -o "Desktop/tux.png" 
  -k "https://0.0.0.0:4443/storage/v1/b/sample-bucket/o/tux.png"
```
and this happened when I tried to upload image file (png, jpeg, ...), with text file it's ok

**Comment by fsouza** [comment]
Are you using the filesystem storage backend? If so, I think this is the same as #296 🤔 Need to investigate it further to check what's going on.

**Comment by nhatmvo** [comment]
No, just a normal image file file from the internet, I used postman for testing, and the content type is returned correctly, so probably it's not the same as the issue you mentioned about.

**Comment by fsouza** [comment]
@nhatmvo oh, I mean which backend within fake-gcs-server. What's the command you used to start fake-gcs-server?

**Comment by nhatmvo** [comment]
I use exactly the command from README.md.

**Comment by fsouza** [comment]
Can you try with `-backend memory` and check if the error persists?

```
docker run -d --name fake-gcs-server -p 4443:4443 fsouza/fake-gcs-server -backend memory
```

**Comment by Sytten** [comment]
I think its because the url is missing a query param correctly, it should be: `https://0.0.0.0:4443/storage/v1/b/sample-bucket/o/tux.png?alt=media`

**Comment by BlackBeardBana** [comment]
Would it be possible to support the same functionality without the need for the query param? `?alt=media`

**Comment by fsouza** [comment]
> Would it be possible to support the same functionality without the need for the query param? ?alt=media

For that, you'd need to use the download URL. You're making a call to the [Objects: get](https://cloud.google.com/storage/docs/json_api/v1/objects/get) API and it requires ?alt=media to serve object data.

**Comment by BlackBeardBana** [comment]
@fsouza
Thanks for the response, what would the download URL look like? 

I can reach the object data through the browser via:

- https://localhost:4443/storage/v1/b/image-bucket/o/folder/image.jpg

and I've tried: 
- https://localhost:4443/image-bucket/folder/image.jpg
- https://localhost:4443/download/storage/v1/b/iimage-bucket/0/folder/image.jpg

to download the image but both endpoints return a 404.

Any help would be appreciated.
